client/executor/wasmi/src/lib.rs

```diff
@@ -295,9 +319,19 @@ impl<'a> wasmi::ModuleImportResolver for Resolver<'a> {
 			}
 		}
 
-		Err(wasmi::Error::Instantiation(
-			format!("Export {} not found", name),
-		))
+		if self.enable_stub {
+			trace!("Could not find function {}, a stub will be provided instead.", name);
+			let id = self.missing_function_id.borrow().clone();
+			self.missing_functions.borrow_mut().insert(id, name.to_string());
```

Comment:
We actually just require `missing_functions` to be a `Vec`.

The index for the function will be computed as:
```
self.host_functions.len() + self.missing_functions.len()
```
(before adding the new missing function name!)

And in invoke we do the following
```
if index >= host_functions.len() {
     Err("{} not found whatever", self.missing_functions[index - host_functions.len())
}
```
```

Reply:
Yes that's what I implemented before as you can see here: https://github.com/paritytech/substrate/pull/4550/commits/233399499bd5cf097370b01f20bb66812dbdeb91

But I told you this looks way too clever and I should probably use a hashmap and you said you would have use a hashmap.

I don't mind reverting it but I'm not in favor of this because I find that it doesn't look like it brings performance improvement, maybe memory, but it certainly make the code harder to understand.

Now that I shared my humble opinion. What do you prefer? :grin:

Reply:
Yeah, I wanted to use a hashmap because I did not thought about this solution :sweat_smile: 
It clearly will improve the performance, as we just need to access the memory by index and not need to look into a hashmap. Please revert, but also make sure that `invoke` does not tries to use an invalid index.

Reply:
done